### PR TITLE
[4.2] Added missing Filesystem tests

### PR DESCRIPTION
**Pull request by manan-jadhav**
Follow up for #6715.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
The cs is still wrong. We need branes on new lies, and a space before we VB open a bracket in if statements and loops.

File: tests/Filesystem/FilesystemTest.php [hunk 1]
```diff
@@ -4,126 +4,316 @@
 
 class FilesystemTest extends PHPUnit_Framework_TestCase {
 
+	static protected $testDirectory = __DIR__;
```
Comment (GrahamCampbell) [inline tests/Filesystem/FilesystemTest.php]: please don't do this

File: tests/Filesystem/FilesystemTest.php [hunk 2]
```diff
@@ -4,126 +4,316 @@
...
+	protected $filesystem = null;
+
+	public function tearDown()
+	{
+		if($this->filesystem == null){
+			$this->filesystem = new Filesystem();
+		}
+		$tempFiles = ['foo','bar','foo.txt','bar.txt','file.txt','file.php','tmp','tmp2'];
+		foreach ($tempFiles as $item) {
+			if($this->filesystem->isDirectory(self::$testDirectory.DIRECTORY_SEPARATOR.$item)){
```
Comment (GrahamCampbell) [inline tests/Filesystem/FilesystemTest.php]: don't use DIRECTORY_SEPARATOR

Comment (CurosMJ) [inline tests/Filesystem/FilesystemTest.php]: Ok.

File: tests/Filesystem/FilesystemTest.php [hunk 3]
```diff
@@ -4,126 +4,318 @@
...
 class FilesystemTest extends PHPUnit_Framework_TestCase {
 
+	static protected $testDirectory = __DIR__;
+
+	protected $filesystem = null;
+
+	public function tearDown()
+	{
+		if ($this->filesystem == null)
+			$this->filesystem = new Filesystem();
```
Comment (GrahamCampbell) [inline tests/Filesystem/FilesystemTest.php]: cs still wrong

File: tests/Filesystem/FilesystemTest.php [hunk 4]
```diff
@@ -4,126 +4,318 @@
...
+		$tempFiles = ['foo','bar','foo.txt','bar.txt','file.txt','file.php','tmp','tmp2'];
+		foreach ($tempFiles as $item) 
+		{
+			if($this->filesystem->isDirectory(self::$testDirectory.'/'.$item))
+			{
+				$this->filesystem->deleteDirectory(self::$testDirectory.'/'.$item);
+			}
+			else if($this->filesystem->isFile(self::$testDirectory.'/'.$item))
```
Comment (GrahamCampbell) [inline tests/Filesystem/FilesystemTest.php]: and here

File: tests/Filesystem/FilesystemTest.php [hunk 5]
```diff
@@ -4,126 +4,318 @@
...
+	protected $filesystem = null;
+
+	public function tearDown()
+	{
+		if ($this->filesystem == null)
+			$this->filesystem = new Filesystem();
+		$tempFiles = ['foo','bar','foo.txt','bar.txt','file.txt','file.php','tmp','tmp2'];
+		foreach ($tempFiles as $item) 
+		{
+			if($this->filesystem->isDirectory(self::$testDirectory.'/'.$item))
```
Comment (GrahamCampbell) [inline tests/Filesystem/FilesystemTest.php]: and here

File: tests/Filesystem/FilesystemTest.php [hunk 6]
```diff
@@ -4,126 +4,318 @@
...
+		$this->assertEquals($time,$files->lastModified(self::$testDirectory.'/foo.txt'));
+		@unlink(self::$testDirectory.'/foo.txt');
+	}
+
+
+	public function testIsWritable()
+	{
+		file_put_contents(self::$testDirectory.'/foo.txt', 'foo');
+		$files = new Filesystem;
+		@chmod(self::$testDirectory.'/foo.txt', 0444);
```
Comment (GrahamCampbell) [inline tests/Filesystem/FilesystemTest.php]: should say static not self given you made the property protected

**Comment by CurosMJ** [comment]
ping @GrahamCampbell . I've done as much as I could, I've used `Filesystem.php` as a guidance for CS. 

If the CS is still incorrect, please fork from my branch and then correct everything and psuh. 

I dont want the time I put in this thing to be wasted.

**Comment by GrahamCampbell** [comment]
Taylor didn't like all the other changes you made. Could you send this without the tear down stuff for now, and with without changing the dir constant stuff please. Also, please send to 5.0 not 4.2.

**Comment by CurosMJ** [comment]
Sure. :+1: 

I first sent this to 5.0 only lol.

**Pull request closed without merging**